src/scene/mesh-instance.js

```diff
@@ -372,6 +373,32 @@ Object.assign(MeshInstance.prototype, {
         // Deprecated
     },
 
+    // test if meshInstance is visible by camera. It requires the frustum of the camera to be up to date, which forward-renderer
+    // takes care of. This function should  not be called elsewhere.
+    _isVisible: function (camera) {
```

Comment:
Just playing devil's advocate here, but I'm wondering if this function should take a `pc.Frustum` instead of a camera.

Reply:
the custom visibility function accepts camera, so that would change the pubic API
this.isVisibleFunc(camera)

Reply:
or at least required larger changes

Reply:
Ooo. Good point. But just to be clear, `pc.Camera` is *not* (currently) public API. And neither is `isVisibleFunc` AFAIK.

Reply:
hey good to know. I looked at how the isVisibleFunc is used by elements, and in 
isVisibleForCamera: function (camera)
they use other properties from camera than the frustum, so I don't think this can change

Reply:
Fair enough. We *might* have to make `pc.Camera` public at some point then. Or alternatively merge `pc.Camera` with `pc.CameraComponent`.